includes/class-mention.php

```diff
@@ -72,16 +72,16 @@ public static function replace_with_links( $result ) {
 			! is_wp_error( $metadata ) &&
 			( ! empty( $metadata['id'] ) || ! empty( $metadata['url'] ) )
 		) {
+			$url = isset( $metadata['url'] ) ? object_to_uri( $metadata['url'] ) : object_to_uri( $metadata['id'] );
+
 			$username = ltrim( $result[0], '@' );
 			if ( ! empty( $metadata['name'] ) ) {
 				$username = $metadata['name'];
 			}
 			if ( ! empty( $metadata['preferredUsername'] ) ) {
-				$username = $metadata['preferredUsername'];
+				$username = sprintf( '%1$s@%2$s', $metadata['preferredUsername'], wp_parse_url( $url, PHP_URL_HOST ) );
 			}
 
-			$url = isset( $metadata['url'] ) ? object_to_uri( $metadata['url'] ) : object_to_uri( $metadata['id'] );
-
 			return \sprintf( '<a rel="mention" class="u-url mention" href="%s">@<span>%s</span></a>', esc_url( $url ), esc_html( $username ) );
```

Comment:
```suggestion
			return \sprintf( '<a rel="mention" class="u-url mention" href="%s">@%s</a>', esc_url( $url ), esc_html( $username ) );
```

There is still a difference in HTML. One is adding the `<span>` and the transformer seems to not add it.